data/web/autodiscover.php

```diff
@@ -1,36 +1,22 @@
 <?php
-require_once 'inc/vars.inc.php';
+require_once 'inc/clientconfig.inc.php';
 require_once 'inc/functions.inc.php';
-$config = array(
-     'useEASforOutlook' => 'yes',
-     'autodiscoverType' => 'activesync',
-     'imap' => array(
-       'server' => $mailcow_hostname,
-       'port' => '993',
-       'ssl' => 'on',
-     ),
-     'smtp' => array(
-       'server' => $mailcow_hostname,
-       'port' => '465',
-       'ssl' => 'on'
-     ),
-     'activesync' => array(
-       'url' => 'https://'.$mailcow_hostname.'/Microsoft-Server-ActiveSync'
-     )
-);
 
+$config = get_client_config();
 if(file_exists('inc/vars.local.inc.php')) {
 	include_once 'inc/vars.local.inc.php';
 }
 
+$config['autodiscoverType'] = 'activesync';
+
 /* ---------- DO NOT MODIFY ANYTHING BEYOND THIS LINE. IGNORE AT YOUR OWN RISK. ---------- */
 
 error_reporting(0);
 
 $data = trim(file_get_contents("php://input"));
 
 // Desktop client needs IMAP, unless it's Outlook 2013 or higher on Windows
-if (strpos($data, 'autodiscover/outlook/responseschema')) { // desktop client
+if (strpos($data, 'autodiscover/outlook/responseschema') !== FALSE) { // desktop client
```

Comment:
Yes, but it doesn't actually matter, it still worked. `strpos` should never be compared in a non-typesafe way as it may return both `0` and `FALSE`.